src/Export/Adapters/OptionsAdapter.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\Shopware6Common\Export\Adapters;
+
+use FINDOLOGIC\Export\Data\Attribute;
+use FINDOLOGIC\Shopware6Common\Export\Config\PluginConfig;
+use FINDOLOGIC\Shopware6Common\Export\Utils\Utils;
+use Vin\ShopwareSdk\Data\Entity\Product\ProductEntity;
+use Vin\ShopwareSdk\Data\Entity\PropertyGroupOption\PropertyGroupOptionCollection;
+use Vin\ShopwareSdk\Data\Entity\PropertyGroupOption\PropertyGroupOptionEntity;
+
+class OptionsAdapter
+{
+
+    protected PluginConfig $pluginConfig;
+
+    public function __construct(
+        PluginConfig $pluginConfig,
+    ) {
+        $this->pluginConfig = $pluginConfig;
+    }
+
+    /**
+     * @return Attribute[]
+     */
+    public function getOptionAttributes(ProductEntity $product): array
+    {
+        if (!$product->options || !$product->options->count()) {
+            return [];
+        }
+
+        return Utils::getPropertyGroupOptionAttributes($product->options, $this->pluginConfig);
```

Comment:
I know I suggested this, but I just realized that a Trait is the only solution. This way, the functions can not be extended within a extension.
With a trait, you can also remove the functions from the `AttributeAdapter`, as there should not be an issue with existing extensions.